### PR TITLE
chore: remove veneer wrappers from bigtable-client-core

### PR DESCRIPTION
**Pull request by igorbernstein2**
The wrappers were previously moved up a layer bigtable-hbase.

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java [hunk 1]
```diff
@@ -45,7 +46,11 @@ public void connectivityTest() {
             .setUserAgent("Test")
             .build();
     try (BigtableSession bs = new BigtableSession(options)) {
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java:48]: nit - `bigtableSession`

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java [hunk 2]
```diff
@@ -45,7 +46,11 @@ public void connectivityTest() {
...
-      bs.getTableAdminClientWrapper().listTables();
+      bs.getTableAdminClient()
+          .listTables(
+              ListTablesRequest.newBuilder()
+                  .setParent("projects/" + projectId + "/instances/" + instanceId)
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java:52]: I would use the `BIGTABLE_V2_INSTANCE_FMT` utility in BigtableInstanceName

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java [hunk 3]
```diff
@@ -288,7 +288,6 @@
 
   /** A flag to decide which implementation to use for data & admin operation */
   public static final String BIGTABLE_USE_GCJ_CLIENT = "google.bigtable.use.gcj.client";
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java:290]: can we remove?

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/batch/common/CloudBigtableServiceImpl.java [hunk 4]
```diff
@@ -29,8 +30,12 @@
   @Override
   public List<KeyOffset> getSampleRowKeys(CloudBigtableTableConfiguration config)
       throws IOException {
-    try (BigtableSession session = new BigtableSession(config.toBigtableOptions())) {
-      return session.getDataClientWrapper().sampleRowKeys(config.getTableId());
+
+    // TODO: figure out how to stick to HBase api here
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/batch/common/CloudBigtableServiceImpl.java:34]: what does this mean exactly? Should we create a tracking issue?

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java (hunk 3 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java:290]: no, its still used by:
https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/BigtableHBaseSettings.java#L43

After beta, when we drop bigtable-client-core, we'll be able to remove it

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/batch/common/CloudBigtableServiceImpl.java (hunk 4 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/batch/common/CloudBigtableServiceImpl.java:34]: This would be part of making bigtable-hbase compatible with HBaseIO.  The work would need to be scoped and this would be a sub-task

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java (hunk 2 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java:52]: done

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java (hunk 1 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/IntegrationBigtableSessionTest.java:48]: done

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java [hunk 5]
```diff
@@ -1,145 +0,0 @@
...
-import java.net.ServerSocket;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
-
-public class TestBigtableSessionConnectionCache {
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java:43]: why is this deleted?

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java [hunk 6]
```diff
@@ -1,140 +0,0 @@
...
-import com.google.api.core.ApiFuture;
-import com.google.api.core.SettableApiFuture;
-import com.google.api.gax.batching.Batcher;
-import com.google.api.gax.batching.BatcherImpl;
-import com.google.api.gax.batching.BatchingDescriptor;
-import com.google.api.gax.batching.BatchingSettings;
-import com.google.api.gax.batching.FlowControlSettings;
-import com.google.api.gax.batching.FlowController.LimitExceededBehavior;
-import com.google.api.gax.rpc.UnaryCallable;
-import com.google.cloud.bigtable.core.IBulkMutation;
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java:35]: is this already covered with a BulkMutation test?

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationWrapper.java [hunk 7]
```diff
@@ -1,90 +0,0 @@
...
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.mockito.Mockito.doNothing;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
-
-import com.google.bigtable.v2.MutateRowResponse;
-import com.google.bigtable.v2.MutateRowsRequest;
-import com.google.cloud.bigtable.core.IBulkMutation;
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationWrapper.java:26]: same q here

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java (hunk 5 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java:43]: This is testing connection pool caching of veneer channels. This doesn't belong in bigtable-client-core. I need to port this to BigtableVeneerApi in bigtable-hbase. I'll revive this in a follow up PR

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java (hunk 6 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java:35]: This testing implementation of a removed wrapper. The equivilent is `TestBulkMutationVeneerApi`

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationWrapper.java (hunk 7 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationWrapper.java:26]: its already covered by TestBulkMutationClassicApi

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java (hunk 5 above)
Comment (kolea2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java:43]: Can we add it as part of this one? I don't want it getting lost in the shuffle.

Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java:43]: It should stand on it own. it should be ready for review shortly

**Review by kolea2 — APPROVED** [review]

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java (hunk 5 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSessionConnectionCache.java:43]: https://github.com/googleapis/java-bigtable-hbase/pull/2916
